frame/beefy/src/lib.rs

```diff
@@ -98,22 +98,27 @@ pub mod pallet {
 impl<T: Config> Pallet<T> {
 	/// Return the current active BEEFY validator set.
 	pub fn validator_set() -> ValidatorSet<T::BeefyId> {
-		ValidatorSet::<T::BeefyId> { validators: Self::authorities(), id: Self::validator_set_id() }
+		let validators: Vec<T::BeefyId> = Self::authorities();
+		let id: beefy_primitives::ValidatorSetId = Self::validator_set_id();
+		// Safe to unwrap() since `Self::authorities()` guaranteed to be non-empty
+		// by Self::initialize_authorities().
+		ValidatorSet::<T::BeefyId>::new(validators, id).unwrap()
 	}
 
 	fn change_authorities(new: Vec<T::BeefyId>, queued: Vec<T::BeefyId>) {
 		// As in GRANDPA, we trigger a validator set change only if the the validator
 		// set has actually changed.
-		if new != Self::authorities() {
+		if new != Self::authorities() && !new.is_empty() {
```

Comment:
What's this additional check about? Under what circumstances would `new` be empty? Is this behavior covered with a test?

Reply:
`new` should never be empty since `change_authorities` is only called from `on_new_session`, **but** this is not enforced by the compiler or anything else, in case of misuse we'd rather do nothing (keep old authorities) then run with no authorities, right?

It's just an extra sanity check.

Reply:
I think we should most importantly follow whatever GRANDPA is doing. If there is a check like that there let's keep it, otherwise I'd be for removing it. We need to make the rest of the code resilient to an empty validator set anyway, because that's the default value of the storage item. @andresilva what's your take on this?

Reply:
If we keep the check though, I'd be for changing the code below a bit to:
```rust
if new != self.authorities() {
  let next_id = Self::validator_set_id() + 1u64;
  if let Some(validator_set) = ValidatorSet::<T::BeefyId>::new(new, next_id) {
      /// ... the rest follows here
  }
}
```

Reply:
done.

Reply:
removed the additional check and staying consistent with grandpa